google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineFacetEditor.java

```diff
@@ -172,6 +174,10 @@ public String getHelpTopic() {
   @Override
   public void onFacetInitialized(@NotNull Facet facet) {
     AppEngineWebIntegration.getInstance().setupDevServer();
+
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.APP_ENGINE_ADD_FACET)
```

Comment:
Well spotted! FacetDetector is deprecated, but we're actually already extending FacetBasedFrameworkDetector, which is the new recommended way. I didn't realise, but we can use the FacetBasedFrameworkDetector.setupFacet() method to track facets added through the Configure popup.

As for ProjectWideFacetListener, I can't figure out how we can use it. It's being used by IJ on a UI component, but I can't really decide on the best place to add it. In any case, since we already have all the metrics we need, I suggest we don't use this, unless you have any suggestions.